src/test/java/org/sqlite/ConnectionTest.java

```diff
@@ -50,13 +50,16 @@ public void executeUpdateOnClosedDB() throws SQLException {
     }
 
     @Test
-    public void readOnly() throws SQLException {
-
+    public void readOnly() throws SQLException, IOException {
         // set read only mode
         SQLiteConfig config = new SQLiteConfig();
         config.setReadOnly(true);
 
         try (Connection conn = DriverManager.getConnection("jdbc:sqlite:", config.toProperties())) {
+            assertThat(((SQLiteConnection) conn).getDatabase().isReadOnly("main")).isTrue();
+            assertThat(((SQLiteConnection) conn).getDatabase().isReadOnly(null)).isTrue();
```

Comment:
what does that mean exactly ? what would be the `null` database, and why is it readonly ?

Reply:
does that mean calling `sqlite3_db_readonly` with `null` will return `1` ?

Reply:
yep, it can, if the main database is read-only. i made sure to hit this case in the tests.

Reply:
i would probably have added a check on the parameter being non-null though. Doesn't make any sense to try with `null`, as it has no semantic meaning.

Reply:
i can add that. i had a null check initially until i was testing and realized that null meant the "first" database to sqlite3, which is also always the "main" database as far as i can tell.

Reply:
well it depends what is the meaning in the sqlite function, if it is a synonym for "main", or if it is something else

Reply:
if we pass in null, sqlite returns the read-only status of the database at index 0 -- the first database, which as far as i can tell always means the "main" database. in effect, i think it is a synonym for "main". that's why i was like, well it doesn't hurt anything to pass in null, and it's less code. it's sloppy/lazy on the caller's part to pass in null, and i prefer being explicit with the database name, so i can add the null check back. can i throw a NullPointerException, or IllegalArgumentException? :-)

Reply:
i am ok with passing null if the underlying sqlite function accepts it and has a meaning for it. We need to put that in the function's comment though, so we we know why we didn't guard about null in the first place

Reply:
ah, ok, i made that comment on `is_read_only`, i can copy the description to the other method's comment too

Reply:
updated with a new commit